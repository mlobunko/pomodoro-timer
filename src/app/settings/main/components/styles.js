import styled from "styled-components";
import ImageBgTomato from "../images/bg-tomato.svg";

export const StyledButtons = styled.div`
  margin-top: 1.2rem;
  margin-bottom: 0.5rem;
  flex-shrink: 0;
  width: 31rem;
  height: 4.1rem;
  display: flex;
  justify-content: center;

  @media (min-width: 360px) {
    width: 35rem;
  }

  @media (min-width: 600px) {
    width: 42rem;
  }
`;
StyledButtons.displayName = "StyledButtons";

export const StyledCenterOfPage = styled.div`
  height: 65%;
  width: 31rem;
  flex: 1 0 auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
  font-size: 3.2rem;

  @media (min-width: 360px) {
    width: 35rem;
  }
  @media (min-width: 600px) {
    width: 42rem;
    height: 80%;
  }
`;
StyledCenterOfPage.displayName = "StyledCenterOfPage";

export const StyledHeader = styled.h1`
  margin: 0;
  flex-shrink: 0;
  text-align: center;
`;
StyledHeader.displayName = "StyledHeader";

export const StyledListOfSettings = styled.div`
  display: grid;
  grid-template-columns: 40% 60%;
  grid-template-rows: 5rem 5rem 5rem 5rem;
  grid-gap: 1rem;
  justify-items: center;
  align-items: center;
`;
StyledListOfSettings.displayName = "StyledListOfSettings";

export const StyledMainComponent = styled.div`
  background: url(${ImageBgTomato}) center;
  background-size: cover;
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  align-items: center;
  font-size: 1.6rem;
  font-family: CarterOne;
`;
StyledMainComponent.displayName = "StyledMainComponent";
