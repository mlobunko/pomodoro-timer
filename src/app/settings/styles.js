import styled from "styled-components";

export const StyledInputs = styled.div`
  display: flex;
  justify-content: "space-evenly";
`;
StyledInputs.displayName = "StyledInputs";

export const StyledInput = styled.div`
  width: 4.5rem;
`;
StyledInput.displayName = "StyledInput";
