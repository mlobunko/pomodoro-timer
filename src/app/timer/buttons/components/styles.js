import styled from "styled-components";

export const StyledButtonsComponent = styled.div`
  margin-top: 1.2rem;
  margin-bottom: 0.5rem;
  flex-shrink: 0;
  width: 31rem;
  height: 4.1rem;
  display: flex;
  justify-content: space-between;

  @media (min-width: 360px) {
    width: 35rem;
  }

  @media (min-width: 600px) {
    width: 42rem;
  }
`;
StyledButtonsComponent.displayName = "StyledButtonsComponent";
