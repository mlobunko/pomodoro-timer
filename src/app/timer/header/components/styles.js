import styled from "styled-components";
import ImageTomatoCool from "shared/images/tomato-cool.svg";

export const StyledHeader = styled.h1`
  margin: 0;
  flex-shrink: 0;
  text-align: center;
  font-family: CarterOne;
`;
StyledHeader.displayName = "StyledHeader";

export const StyledInfo = styled.div`
  display: inline-block;
  margin-left: 1rem;
  font-size: 2.5rem;
`;
StyledInfo.displayName = "StyledInfo";

export const StyledTomatoIcon = styled.div`
  display: inline-block;
  background: url(${ImageTomatoCool}) no-repeat center center;
  margin-left: 0.2rem;
  background-size: cover;
  width: 4rem;
  height: 4.2rem;
`;
StyledTomatoIcon.displayName = "StyledTomatoIcon";
