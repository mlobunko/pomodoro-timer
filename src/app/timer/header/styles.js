import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import styled from "styled-components";

export const StyledFontAwesomeIcon = styled(FontAwesomeIcon)`
  color: rgb(61, 153, 112);
`;
StyledFontAwesomeIcon.displayName = "StyledFontAwesomeIcon";
