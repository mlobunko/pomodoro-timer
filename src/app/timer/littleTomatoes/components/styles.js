import styled, { keyframes } from "styled-components";

import ImageTomato from "shared/images/tomato.svg";

const styleTomato = `
  background: url(${ImageTomato}) no-repeat center;
  width: 70%;
`;

const blinking = keyframes`
  0% {
    opacity: 0;
  }
  50% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
`;

export const StyledAnimatedLittleTomato = styled.div`
  ${styleTomato};
  animation: ${blinking} 3s linear infinite;
`;
StyledAnimatedLittleTomato.displayName = "StyledAnimatedLittleTomato";

export const StyledLittleTomato = styled.div`
  ${styleTomato};
`;
StyledLittleTomato.displayName = "StyledLittleTomato";

export const StyledLittleTomatoesComponent = styled.div`
  display: grid;
  grid-template-columns: 25% 25% 25% 25%;
  grid-template-rows: 7.5rem;
  justify-items: center;
`;
StyledLittleTomatoesComponent.displayName = "StyledLittleTomatoesComponent";
