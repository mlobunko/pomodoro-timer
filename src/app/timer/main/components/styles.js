import styled from "styled-components";
import ImageBgClock from "../images/bg-clock.svg";

export const StyledBackground = styled.div`
  position: fixed;
  left: 0;
  right: 0;
  z-index: -1;
  background: url(${ImageBgClock});
  background-size: cover;
  height: 100vh;
  filter: opacity(15%);
`;

export const StyledCenterOfPage = styled.div`
  height: 65%;
  width: 31rem;
  flex: 1 0 auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
  font-size: 3.2rem;

  @media (min-width: 360px) {
    width: 35rem;
  }
  @media (min-width: 600px) {
    width: 42rem;
    height: 80%;
  }
`;

export const StyledFront = styled.div`
  z-index: 2;
`;

export const StyledPage = styled.div`
  font-size: 1.6rem;
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  align-items: center;
`;
