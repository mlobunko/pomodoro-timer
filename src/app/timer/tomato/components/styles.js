import styled from "styled-components";
import PropTypes from "prop-types";

import ImageTomato from "shared/images/tomato.svg";
import ImageTomatoCool from "shared/images/tomato-cool.svg";

export const StyledTomatoComponent = styled.div`
  background: ${props =>
    props.isWorkingTime
      ? `url(${ImageTomato}) no-repeat center center`
      : `url(${ImageTomatoCool}) no-repeat center center`};
  background-size: contain;
  width: 100%;
  height: 30rem;
  line-height: 38rem;
  font-size: 7.5rem;
  color: white;
  text-align: center;
  font-family: TitanOne;

  @media (min-width: 360px) {
    height: 34rem;
    line-height: 43rem;
  }

  @media (min-width: 600px) {
    height: 38rem;
    line-height: 48rem;
    font-size: 9.5rem;
  }
`;

StyledTomatoComponent.displayName = "StyledTomatoComponent";

StyledTomatoComponent.propTypes = {
  isWorkingTime: PropTypes.bool
};
