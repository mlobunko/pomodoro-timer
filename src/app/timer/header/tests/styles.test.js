import React from "react";
import { shallow } from "enzyme";
import "enzyme-to-json";

import { StyledFontAwesomeIcon } from "../styles";

describe("StyledFontAwesomeIcon", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<StyledFontAwesomeIcon />);
    expect(wrapper).toMatchSnapshot();
  });
});
