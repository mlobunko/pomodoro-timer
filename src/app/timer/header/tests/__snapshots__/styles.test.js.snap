// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledFontAwesomeIcon renders correctly 1`] = `
<FontAwesomeIcon
  border={false}
  className="sc-bdVaJa eXMIgU"
  fixedWidth={false}
  flip={null}
  icon={null}
  inverse={false}
  listItem={false}
  mask={null}
  pull={null}
  pulse={false}
  rotation={null}
  size={null}
  spin={false}
  symbol={false}
  transform={null}
/>
`;
